task-driver/src/helpers.rs

```diff
@@ -337,44 +335,39 @@ pub(crate) async fn update_wallet_validity_proofs(
     Ok(())
 }
 
-/// Attach a copy of the witness to the locally managed state
-///
-/// This witness is referenced by match computations to provide linkable
-/// commitments into a previously proven validity bundle
-async fn record_validity_witness(
-    order_id: &OrderIdentifier,
-    commitments_witness: &SizedValidCommitmentsWitness,
-    valid_reblind_witness: &SizedValidReblindWitness,
-    global_state: &RelayerState,
-) {
-    let witness_bundle = OrderValidityWitnessBundle {
-        reblind_witness: Arc::new(valid_reblind_witness.clone()),
-        commitment_witness: Arc::new(commitments_witness.clone()),
-    };
-
-    global_state
-        .read_order_book()
-        .await
-        .attach_validity_proof_witness(order_id, witness_bundle)
-        .await;
-}
-
-/// Attach a validity proof to the locally managed state
+/// Attach a validity proof and witness to the locally managed state
 ///
 /// This is gossipped to the network so that peers may verify the validity
 /// bundle and schedule matches on the proven order
-pub(crate) async fn record_validity_proof(
+async fn record_validity_proof_and_witness(
```

Comment:
nit: can we change the name (and certainly doc comment) of this method to indicate that it also computes a linking proof? for the name, maybe something like `record_linked_validity_proof_and_witness`, or `link_and_record_validity_bundle`